app/src/play/java/de/danoeh/antennapod/dialog/RatingDialog.java

```diff
@@ -0,0 +1,116 @@
+package de.danoeh.antennapod.dialog;
+
+import android.app.Activity;
+
+import android.content.Context;
+import android.content.SharedPreferences;
+
+import androidx.annotation.VisibleForTesting;
+import android.util.Log;
+import android.widget.Toast;
+
+import java.lang.ref.WeakReference;
+import java.util.concurrent.TimeUnit;
+
+import com.google.android.play.core.review.ReviewInfo;
+import com.google.android.play.core.review.ReviewManager;
+import com.google.android.play.core.review.ReviewManagerFactory;
+import com.google.android.play.core.tasks.Task;
+
+public class RatingDialog {
+
+    private RatingDialog(){
+    }
+
+    private static final String TAG = RatingDialog.class.getSimpleName();
+    private static final int AFTER_DAYS = 7;
+
+    private static WeakReference<Context> mContext;
+    private static SharedPreferences mPreferences;
+
+    private static final String PREFS_NAME = "RatingPrefs";
+    private static final String KEY_RATED = "KEY_WAS_RATED";
+    private static final String KEY_FIRST_START_DATE = "KEY_FIRST_HIT_DATE";
+
+    public static void init(Context context) {
+        mContext = new WeakReference<>(context);
+        mPreferences = context.getSharedPreferences(PREFS_NAME, Context.MODE_PRIVATE);
+
+        long firstDate = mPreferences.getLong(KEY_FIRST_START_DATE, 0);
+        if (firstDate == 0) {
+            resetStartDate();
+        }
+    }
+
+    public static void check() {
+        if (shouldShow()) {
+            try {
+                showInAppReview();
+            } catch (Exception e) {
+                Log.e(TAG, Log.getStackTraceString(e));
+            }
+        }
+    }
+
+    private static void showInAppReview() {
+        Context context = mContext.get();
+        if (context == null) {
+            return;
+        }
+
+        //ReviewManager manager = new FakeReviewManager(context);
+        ReviewManager manager = ReviewManagerFactory.create(context);
+        Task<ReviewInfo> request = manager.requestReviewFlow();
+
+        request
+                .addOnCompleteListener(task -> {
+                    if (task.isSuccessful()) {
+                        ReviewInfo reviewInfo = task.getResult();
+                        Task<Void> flow = manager.launchReviewFlow((Activity) context, reviewInfo);
+                        flow
+                                .addOnCompleteListener(task1 -> {
+                                    saveRated();
+                                    Toast.makeText(context, "Review saved", Toast.LENGTH_SHORT).show();
```

Comment:
I think users know that. No need to display the toast. Also, it is not really correct since the same code path is also executed when the review is cancelled.